### ISSUE TITLE
Folding of record fields

### ISSUE THREAD

**Issue by anton-trunov** [ProofGeneral/PG#367]
Let's consider the following snippet:
```coq
Record foo := mkFoo { unfoo : nat }.
Goal
mkFoo ( 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 ) 
=
mkFoo ( 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 ) 
.
```

I've noticed that PG offers me to fold/unfold the values of the record fields (is this intended?). But when I try to fold the second occurence of `unfoo` in the goal, PG actually folds the first `unfoo`.

Also, it looses `|}` and `{|` (see the picture below).

- Before folding:
```
   {|
-  unfoo := ... + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 +
            0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 |} =
   {|
-  unfoo := ... + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 +
            0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 |}
```

- After folding the second `unfoo` (I click on the second `-`):

```
   {|
-  unfoo := ... =
-  unfoo := ... + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 +
           0 + 0 + 0 + 0 + 0 + 0 + 0 + 0 |}
```

**Comment by Matafou** [comment]
No it is not intended. It is a bug. Only hypothesis should be foldable. I will look at it.

Hit "H" (shift-h) to unfold everything.

**Comment by Matafou** [comment]
It should be fixed by commit c7059c8f. please check and reopen if needed.

**Comment by anton-trunov** [comment]
@Matafou Thanks for the fix, it works!
